app/controllers/api/v1/migrations/external_controller.rb

```diff
@@ -21,26 +21,41 @@ class ExternalController < ApiController
         end
 
         # POST /api/v1/migrations/roles.json
-        # Expects: { role: { :name } }
+        # Expects:
+        # { role: { :name,
+        #            role_permissions: { :CreateRoom, :CanRecord, :ManageUsers, :ManageRoles, :ManageRooms, :ManageRecordings, :ManageSiteSettings }}}
         # Returns: { data: Array[serializable objects(recordings)] , errors: Array[String] }
         # Does: Creates a role.
         def create_role
           role_hash = role_params.to_h
 
           return render_data status: :created if Role.exists? name: role_hash[:name], provider: 'greenlight'
 
-          role = Role.new role_hash.merge(provider: 'greenlight')
+          role = Role.new(name: role_hash[:name], provider: 'greenlight')
 
           return render_error status: :bad_request unless role.save
 
+          # Returns unless the Role has a RolePermission that differs from V3 default RolePermissions values
+          return render_data status: :created unless role_hash[:role_permissions].any?
+
+          # Finds & Updates the associated RolePermissions associated with the Role
+          role_permissions = RolePermission.where(role_id: role.id).includes(:permission)
+          # The role_permissions hash contains only the permissions with value that differs from V3 default values
+          role_hash[:role_permissions].any? && role_hash[:role_permissions].each do |name, value|
+            role_permission = role_permissions.find_by('permissions.name': name)
+            return render_error status: :bad_request unless role_permission
+
+            role_permission.update!(value:)
+            return render_error status: :bad_request unless role_permission.save
```

Comment:
This will result in a ton of queries. Can we not do 1 find and 1 update_all?  See: https://api.rubyonrails.org/v3.2.0/classes/ActiveRecord/Relation.html#method-i-update

Reply:
You're also doing an `update` then a `save` which doesn't really make sense

Reply:
Right, using `update_all` would be great.